src/transformStateWithClones.js

```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let currentState = state;
+  const stateHistory = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      const newState = { ...currentState };
+
+      for (let j = 0; j < action.keysToRemove.length; j++) {
+        const key = action.keysToRemove[j];
+
+        delete newState[key];
+      }
+      currentState = newState;
+    }
+
+    stateHistory.push(currentState);
```

Comment:
You are pushing `currentState` directly to `stateHistory`. This will result in all elements of `stateHistory` referencing the same object. You should push a copy of `currentState` instead to maintain a history of independent state objects.